### PR TITLE
Adds latex, pdf, usfm rendering support, makes some code a bit more consistent, etc.

### PR DESCRIPTION
**Pull request by rbnswartz**
Adds
- PDF support
- Latex support
- USFM support
- Better error message handling
- Some more consinstency in internal behavior

### CODE REVIEW

File: ScriptureRenderingPipeline/Rendering.cs [hunk 1]
```diff
@@ -26,68 +28,202 @@ public static class Rendering
...
+                    if (string.IsNullOrEmpty(fontMappingUrl))
+                    {
+                        return GenerateErrorAndLog($"Latex conversion was requested but font lookup file url was not specified", log, 400);
+                    }
+
+                    // Find out which font we should use in the latex file
+                    var fonts = await GetFonts(fontMappingUrl);
+                    var config = CreateLatexConfig(req.Query);
+                    config.Font = SelectFontForDocument(document, fonts);
+                    log.LogInformation($"Selected {config.Font} for rendering");
```
Comment (PurpleGuitar) [inline ScriptureRenderingPipeline/Rendering.cs:186]: This font selection code looks identical between pdf and latex.  Consider pulling it out into a function.

File: ScriptureRenderingPipeline/Rendering.cs [hunk 2]
```diff
@@ -26,68 +28,202 @@ public static class Rendering
...
-            }
 
-            var output = renderer.Render(document);
-            string outputFilePath = Path.Join(repoDir, "output.docx");
-            using (var stream = new FileStream(outputFilePath, FileMode.Create))
-            {
-                output.Write(stream);
+                if (fileType == "pdf")
+                {
+                    var latexConverterUrl = Environment.GetEnvironmentVariable("PDFConversionEndpoint");
```
Comment (PurpleGuitar) [inline ScriptureRenderingPipeline/Rendering.cs:141]: Should this be named pdfConverterUrl?

File: ScriptureRenderingPipeline/Rendering.cs [hunk 3]
```diff
@@ -123,15 +260,15 @@ private static DocxConfig CreateConfig(IQueryCollection query)
...
                     config.lineSpacing = lineSpacing;
                 }
             }
 
             if (query.ContainsKey("direction"))
             {
-                if(query["direction"] == "rtl")
+                if (query["direction"] == "rtl")
                 {
                     config.textDirection = NPOI.OpenXmlFormats.Wordprocessing.ST_TextDirection.tbRl;
```
Comment (PurpleGuitar) [inline ScriptureRenderingPipeline/Rendering.cs:273]: The method for specifying RTL has changed and no longer depends on the details of NPOI.  Set `config.rightToLeft` to `true`.  More details here: https://github.com/WycliffeAssociates/USFMToolsSharp.Renderers.Docx/blob/master/USFMToolsSharp.Renderers.Docx/DocxConfig.cs

Comment (rbnswartz) [reply ScriptureRenderingPipeline/Rendering.cs:273]: Done

File: ScriptureRenderingPipeline/Rendering.cs (hunk 2 above)
Comment (rbnswartz) [reply ScriptureRenderingPipeline/Rendering.cs:141]: Possibly but I think we'll just backlog it as it is existing code already

**Pull request merged**